File: nomad-samples/python/Content Admin API/content/add_custom_properties.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

def add_custom_properties(AUTH_TOKEN: str, ID: str) -> dict:

    # Check for valid parameters
    if (not AUTH_TOKEN):
        raise Exception("Authentication Token: The authentication token is invalid")

    API_URL = ADMIN_API_URL + "/admin/asset/" + ID
        
    # Create header for the request
    HEADERS = {
  	    "Authorization": "Bearer " + AUTH_TOKEN,
        "Content-Type": "application/json"
    }

    # Build the payload BODY
    BODY = {
        "displayName": "new name goes here",
        "customProperties": {
            "customProp1": "new value",
          	"customProp2": None
        }
    }

    try:
        # Send the request
        RESPONSE = requests.patch(API_URL, headers= HEADERS, data= json.dumps(BODY))

        if RESPONSE.status_code != 200:
            raise Exception("Response returned " + str(RESPONSE.status_code))

        return json.loads(RESPONSE.text)

    except:
        api_exception_handler(RESPONSE, "add custom properties failed")

File: nomad-samples/js/content-admin-api/content/add-custom-properties.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function addCustomProperties(AUTH_TOKEN, ID, NAME, CUSTOM_PROPERTY_NAMES, CUSTOM_PROPERTIES) 
{
		// Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);
  
    const BODY = {
        displayName: NAME,
        customProperties: {
        }
    };

    for (let numProps = 0; numProps < CUSTOM_PROPERTIES.length; ++numProps)
    {
        BODY.customProperties[CUSTOM_PROPERTY_NAMES[numProps]] = CUSTOM_PROPERTIES[numProps];
    }

    // Post
    const RESPONSE = await fetch(`${prjConstants.ADMIN_API_URL}/admin/asset/${ID}`, {
        method: "PATCH",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    }).catch((exception) => {
        throw exception;
    });

    // Check for success
    if (RESPONSE.ok) {
        // Get the JSON from the response
        const INFO = await RESPONSE.json();

        return INFO;
    }
		
  	// There was an error
    await apiExceptionHandler(RESPONSE, "Add custom properties failed");
}